mailbox/backup/src/main/java/org/apache/james/mailbox/backup/zip/MailAccountZipIterator.java

```diff
@@ -91,17 +95,75 @@ private String getMailboxName(ZipEntry current) {
         return current.getName().substring(0, current.getName().length() - 1);
     }
 
-    private MailArchiveEntry fromMailboxEntry(ZipEntry current) throws ZipException {
-        return new MailboxWithAnnotationsArchiveEntry(getMailboxName(current), getMailBoxId(current).get(), NO_ANNOTATION);
+    private MailArchiveEntry fromMailboxEntry(ZipEntryWithContent current, Optional<ZipEntry> nextZipEntry) throws ZipException {
+        ZipEntry entry = current.getEntry();
+        if (isLastEntryOrHasNoAnnotationsDirectory(nextZipEntry)) {
+            //no annotation directory or end of iterator => get Mailbox
+            currentMailBox = Optional.empty();
+            return new MailboxWithAnnotationsArchiveEntry(getMailboxName(entry), getMailBoxId(entry).get(), NO_ANNOTATION);
+        } else {
+            // annotation directory => save current mailbox and fetch next entry
+            currentMailBox = Optional.of(new MailboxWithAnnotationsArchiveEntry(getMailboxName(entry), getMailBoxId(entry).get(), NO_ANNOTATION));
+            return next();
+        }
+    }
+
+    private boolean isLastEntryOrNextEntryIsNotOfType(Optional<ZipEntry> nextZipEntry, ZipEntryType mailboxAnnotationDir) throws ZipException {
+        return !nextZipEntry.isPresent() || !ExtraFieldExtractor.getEntryType(nextZipEntry.get()).equals(Optional.of(mailboxAnnotationDir));
```

Comment:
using directly `Optional.of()` could leads to NullPointerException

Reply:
In this case the ZipEntryType is a value we create so we are confident it's not null. 
But your comment leads me to refactor this method. Thanks !